src/main.cpp

```diff
@@ -227,6 +244,8 @@ namespace
 					move_dir += vec3::constants::backward;
 				if (win.key('d', 1073741903))
 					move_dir += vec3::constants::right;
+				if (win.key(27))
+					exit(EXIT_SUCCESS);
```

Comment:
Two things here:

1. The C standard library (or at least, the vast majority of it) is inside the `std` namespace when used in C++ mode. Any time you call a C standard library function in C++, it must always be prefixed with the `std::` namespace qualifier. Absolutely always, no exceptions. Anyone who tells you it's fine to use the C-style mode is wrong. That it "works" without properly namespace-qualifying it doing so is unfortunate, but a historical unavoidability, and using it like that will eventually cause you to run into issues with C++'s [ADL](https://quuxplusone.github.io/blog/2019/04/08/adl-insanity/) or accidental member function calls at some point.
2. That's moot, however, because the bigger problem is: Friends don't let friends use [`std::exit()`](https://en.cppreference.com/w/cpp/utility/program/exit) 😅  It does what it does in C: terminates the program literally at the callsite. C++ has destructors, though, and terminating from a specific callsite in this manner means they don't run, which is a recipe for disaster. Fine in C because it doesn't have destructors, but in C++ many programs will use them to write files, flush buffers, close file handles, et cetera, and disabling all of that can cause big problems for the host operating system, other programs, or even just subsequent runs of your program.

The proper way to handle point 2 is to just let the program reach a `main()` return statement naturally, which is exactly what will happen here if you return from `run()`. This is what I do myself at line 87:

https://github.com/marzer/rt/blob/5f7500ea7db6c0865d4c50d837787dde5710650e/src/main.cpp#L81-L88

(you can also use [`std::atexit()`](https://en.cppreference.com/w/cpp/utility/program/atexit) to inject a std::exit() handler but that's a kludgy hack and should be avoided.)